bin/plot_truncated_iterations

```diff
@@ -0,0 +1,374 @@
+            and classifications['no steady state'] == 0)
+
+
+def all_nss(classifications):
+    """Return True if all pexecs in a detailed classification dict are 'no steady state'."""
+
+    return (classifications['warmup'] == 0 and classifications['slowdown'] == 0 and
+            classifications['flat'] == 0)
+
+
+def any_nss(classifications):
+    """Return True if any pexec in a detailed classification dict is 'no steady state'."""
+
+    return classifications['no steady state'] > 0
+
+
+def summarise_truncated(data_dir, jsonfile):
+    """Summarise classifications in truncated data that match the original."""
+
+    classifiers = dict()
+    original_results = None
+    truncated_summaries = dict()
+    n_files = len(glob.glob(os.path.join(data_dir, '*_changepoints.json.bz2')))
+    for index, filename in enumerate(glob.glob(os.path.join(data_dir, '*_changepoints.json.bz2'))):
+        if 'truncated_' not in filename:  # Original results file.
+            print('Loading original results. File %d of %d' % (index + 1, n_files))
+            classifiers[DEFAULT_ITERS], original_results = parse_krun_file_with_changepoints([filename])
+            truncated_summaries[DEFAULT_ITERS] = collect_summary_statistics(original_results,
+                                                     classifiers[DEFAULT_ITERS]['delta'], classifiers[DEFAULT_ITERS]['steady'])
+        else:
+            last_iter = int(os.path.basename(filename).split('.')[0].split('_')[1])
+            print('Loading results file truncated to %d iterations. File %d of %d.' %  (last_iter, index + 1, n_files))
+            classifiers[last_iter], truncated_results = parse_krun_file_with_changepoints([filename])
+            truncated_summaries[last_iter] = collect_summary_statistics(truncated_results,
+                                                 classifiers[last_iter]['delta'], classifiers[last_iter]['steady'])
+            del truncated_results
+    assert original_results is not None, 'No original results file.'
+    assert len(original_results.keys()) == 1, 'Expected one machine per results file.'
+    machine = original_results.keys()[0]
+    # Generate CIs for DEFAULT_ITER classification data.
+    class_cis = dict()
+    for key in original_results[machine]['classifications']:
+        if len(original_results[machine]['classifications'][key]) == 0:  # Skipped benchmark.
+            continue
+        class_counts = [original_results[machine]['classifications'][key].count(category) for category in CATEGORIES]
+        class_cis[key] = numpy.array(MCI.multinomialCI(rpy2.robjects.FloatVector(class_counts), ALPHA))
+    # Generate summary of results to be plotted.
+    summary = {'classifications': dict(), 'steady_iter': dict(), 'steady_state_time': dict(), 'intersection': dict()}
+    for last_iter in truncated_summaries:
+        if last_iter not in summary:
+            summary['classifications'][last_iter] = { 'same': 0, 'different': 0 }
```

Comment:
One thing I'd like to do in this function -- which is fairly easy, but will also help performance a fair bit -- is to change all the strings to constants e.g. `same` -> `SAME` (or whatever). In fact for the dictionaries which just have two keys, we'd be much better making them lists (e.g. `[<SAME>, <DIFFERENT>]`) which should also save a lot of memory (which we really need to do).